src/read.rs

```diff
@@ -111,12 +111,10 @@ impl<R> ReadExt for R
 
     #[inline(always)]
     fn read_bytes(&mut self, n: usize) -> io::Result<Vec<u8>> {
-        // We allocate a runtime fixed size buffer, and we are going to read
-        // into it, so zeroing or filling the buffer is a waste. This method
-        // is safe, because the contents of the buffer are only exposed when
-        // they have been overwritten completely by the read.
-        let mut buf = Vec::with_capacity(n);
-        unsafe { buf.set_len(n); }
+        // This method is called with n = 4 bytes for WAVE header data.
+        // Should not worry about initialization overhead with 4 bytes.
+        let mut buf = Vec::<u8>::with_capacity(n);
+        buf.resize(n, 0);
```

Comment:
For `read_bytes`, since the only two calls to it read 4 bytes anyway, we could also delete it and add something like `read_4_bytes` that returns `[u8; 4]`, then we can also do away with the `Vec`, we can just make a buffer initialized to `[0_u8; 4]` on the stack and read into it.